### PR TITLE
issues/170: Fixed message serial out of sync after recover

### PR DESCRIPTION
**Pull request by trenouf**
Also protocol message logging.

### CODE REVIEW

**Comment by trenouf** [comment]
This change might still be wrong. Paddy and I are discussing.

**Comment by trenouf** [comment]
What do you reckon to that Paddy?

File: lib/src/main/java/io/ably/lib/transport/ConnectionManager.java [hunk 1]
```diff
@@ -271,6 +272,7 @@ public void run() {
 	 ***************************************/
 
 	public void onMessage(ProtocolMessage message) throws AblyException {
+		Log.v(TAG, "onMessage(): " + new String(ProtocolSerializer.writeJSON(message)));
```
Comment (paddybyers) [inline lib/src/main/java/io/ably/lib/transport/ConnectionManager.java]: I don't like the way that this serialises every message even though the log level doesn't require it to be emitted. I think it has to be done conditionally on the log level, or the log API has to take a format string and arguments so expansion is under control of the logger.

File: lib/src/main/java/io/ably/lib/transport/ConnectionManager.java [hunk 2]
```diff
@@ -860,6 +860,20 @@ public synchronized void nack(long serial, int count, ErrorInfo reason) {
...
 		}
+
+		/**
+		 * reset the pending message queue, failing any currently pending messages.
+		 * Used when a resume fails and we get a different connection id.
+		 * @param oldMsgSerial the next message serial number for the old
+		 *		connection, and thus one more than the highest message serial
+		 *		in the queue.
+		 */
+		public synchronized void reset(long oldMsgSerial) {
```
Comment (paddybyers) [inline lib/src/main/java/io/ably/lib/transport/ConnectionManager.java]: I think `reset()` should take an `ErrorInfo` argument and pass that to the nack, and then the specific error is determined by whoever calls `reset()`. There will be different situations that will require different error messages/codes.

**Comment by trenouf** [comment]
@paddy fixed your comments.

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**